DoSomething/gambit
# V8 profiler

#### What's this PR do?
It adds the ability to take Heap Snapshots of the app at determined intervals. 

#### How should this be reviewed?
- Install Gambit locally
- checkout this branch `v8-profiler`
- Add the following env variables:
    - AWS_ACCESS_KEY_ID=**Look in lastpass Gambit S3 secured document**
    - AWS_SECRET_ACCESS_KEY=**Look in lastpass Gambit S3 secured document**
    - V8_PROFILER_NAME_PREFIX=local.v8.heap.snapshot
    - V8_PROFILER_ENABLED=true
    - V8_PROFILER_DEFAULT_INTERVAL_MS=20000. **This will create a snapshot every 20 seconds.**
- A log message should confirm the upload was successful:
    - `10:28:38 AM web.1 |  debug: Snapshot upload successful. File name: v8.heap.snapshot_worker1_1491316082756.heapsnapshot`
- If you have s3 access: Confirm that you see a new snapshot created at: [S3 v8-profiler Bucket](https://console.aws.amazon.com/s3/buckets/v8-profiler.dosomething.org/?region=us-east-1&tab=overview)
#### Any background context you want to provide?
We have a memory leak in Gambit that has not been able to be reproduced locally. After trying a lot of unsuccessful ways to reproduce locally, input from the #devops team was required. @pris insights were to invest some time and build the capability of taking Heap Snapshots into Gambit so we could analyze the Heap Snapshot offline. As an added bonus, we can keep this feature handy and turn it off/on as we need via env variables.

#### Relevant tickets
Fixes #837 

#### Checklist
